doc/hma-ui.md

```diff
@@ -0,0 +1,280 @@
+3. **Setup Tx API**: Configures real ThreatExchange integration (but needs credentials for live data)
+
+#### **“Setup Sample API”**
+
+The Sample API (StaticSampleSignalExchangeAPI) is a testing and demonstration tool that provides pre-generated sample hash data for development and evaluation purposes.
+
+##### Development Testing Tool
+
+* Purpose: Provides a mock signal exchange that simulates real threat intelligence feeds without requiring external API credentials  
+* No Authentication Required: Unlike real exchanges (NCMEC, GIFCT, ThreatExchange), it doesn't need API keys or credentials  
+* Instant Setup: Creates sample data immediately for testing the system
+
+##### Sample Data Provider
+
+When you click "Setup Sample API" it:
+
+1. Creates the Exchange: Sets up SEED\_SAMPLE exchange configuration  
+1. Provides Sample Hashes: Returns a fixed set of example hashes for both PDQ (images) and MD5 (videos)  
+1. Populates the System: Gives you 18 sample records initially (as seen in logs: "fetch\_iter() with 18 new records")  
+1. Enables Matching: Allows you to test the matching functionality immediately
+
+##### Fetching Simulation
+
+From the logs, you can see it working:
+
+```py
+INFO in fetcher: SEED_SAMPLE[sample] Fetching signals for SEED_SAMPLE from sample
+INFO in fetcher: SEED_SAMPLE[sample] No checkpoint, should be the first fetch.
+DEBUG in fetcher: SEED_SAMPLE[sample] fetch_iter() with 18 new records
+INFO in fetcher: SEED_SAMPLE[sample] Fetched all data! Up to date!
+```
+
+#### **Why a Sample API?**
+
+##### Development Workflow
+
+* When you're developing or evaluating HMA, you need hash data to test matching  
+* Instead of configuring complex external APIs, you click "Setup Sample API"  
+* It immediately provides workable test data
+
+##### Demo Purpose
+
+* **Shows System Functionality:** Demonstrates how external exchanges work without real credentials  
+* **Testing Matching**: Provides known hashes you can test matching against  
+* **Index Building**: Triggers the indexing process with sample data  
+* **UI Validation:** Lets you see how the exchange status, fetching, and matching systems work
+
+##### Educational Value
+
+* **Example Implementation:** Shows how signal exchanges are structured  
+* **API Pattern**: Demonstrates the exchange API interface without complexity  
+* **No External Dependencies:** Works completely offline
+
+#### **What You See in the UI:**
+
+1. **Before Setup:** Button shows "Setup Sample API"  
+2. **After Setup:** Button shows "✅ Sample API created" and is disabled  
+3. **Exchange Status:** Shows SEED\_SAMPLE exchange in the exchanges page  
+4. **Index Building:** Triggers automatic background indexing of the sample data  
+5. **Matching Ready:** System now has data to match against
+
+### Technical Implementation:
+
+The StaticSampleSignalExchangeAPI is essentially a hardcoded data source that:
+
+* Returns a fixed set of sample hashes (PDQ and MD5)  
```

Comment:
As someone reading this, my question would be: Where do these sample hashes come from? How do I find the images that would trigger matches on these sample hashes?